azblob/zt_url_block_blob_test.go

```diff
@@ -898,3 +899,149 @@ func (s *aztestsSuite) TestBlobPutBlockListModifyBlob(c *chk.C) {
+	c.Assert(resp.Response().Header.Get("x-ms-version"), chk.Equals, ServiceVersion)
+	c.Assert(resp.Response().Header.Get("x-ms-version-id"), chk.NotNil)
+}
+
+func (s *aztestsSuite) TestCreateBlockBlobReturnsVID(c *chk.C) {
+	bsu := getBSU()
+	containerURL, _ := createNewContainer(c, bsu)
+	defer delContainer(c, containerURL)
+
+	testSize := 2 * 1024 * 1024 // 1MB
+	r, _ := getRandomDataAndReader(testSize)
+	ctx := context.Background() // Use default Background context
+	blobURL := containerURL.NewBlockBlobURL(generateBlobName())
+
+	// Prepare source blob for copy.
+	uploadResp, err := blobURL.Upload(ctx, r, BlobHTTPHeaders{}, Metadata{}, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(uploadResp.Response().StatusCode, chk.Equals, 201)
+	c.Assert(uploadResp.rawResponse.Header.Get("x-ms-version"), chk.Equals, ServiceVersion)
+	c.Assert(uploadResp.Response().Header.Get("x-ms-version-id"), chk.NotNil)
+
+	csResp, err := blobURL.CreateSnapshot(ctx, Metadata{}, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(csResp.Response().StatusCode, chk.Equals, 201)
+	c.Assert(csResp.Response().Header.Get("x-ms-version-id"), chk.NotNil)
+
+	listBlobResp, err := containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Snapshots: true}})
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobResp.rawResponse.Header.Get("x-ms-request-id"), chk.NotNil)
+	if len(listBlobResp.Segment.BlobItems) < 2 {
+		c.Fail()
+	}
+
+	deleteResp, err := blobURL.Delete(ctx, DeleteSnapshotsOptionOnly, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(deleteResp.Response().StatusCode, chk.Equals, 202)
+	c.Assert(deleteResp.Response().Header.Get("x-ms-version-id"), chk.NotNil)
+
+	listBlobResp, err = containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Snapshots: true, Versions: true}})
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobResp.rawResponse.Header.Get("x-ms-request-id"), chk.NotNil)
+	if len(listBlobResp.Segment.BlobItems) == 0 {
+		c.Fail()
+	}
+	blobs := listBlobResp.Segment.BlobItems
+	c.Assert(blobs[0].Snapshot, chk.Equals, "")
+}
+
+func (s *aztestsSuite) TestPutBlockListReturnsVID(c *chk.C) {
+	blockIDIntToBase64 := func(blockID int) string {
+		binaryBlockID := (&[4]byte{})[:]
+		binary.LittleEndian.PutUint32(binaryBlockID, uint32(blockID))
+		return base64.StdEncoding.EncodeToString(binaryBlockID)
+	}
+	bsu := getBSU()
+	containerURL, _ := createNewContainer(c, bsu)
+	defer delContainer(c, containerURL)
+
+	blobURL := containerURL.NewBlockBlobURL(generateBlobName())
+
+	data := []string{"Azure ", "Storage ", "Block ", "Blob."}
+	base64BlockIDs := make([]string, len(data))
+
+	for index, d := range data {
+		base64BlockIDs[index] = blockIDIntToBase64(index)
+		resp, err := blobURL.StageBlock(ctx, base64BlockIDs[index], strings.NewReader(d), LeaseAccessConditions{}, nil)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
>log.Fatal(err) [](start = 3, length = 14)

let's not use log.fatal, we can use c.Fail instead

Reply:
Done.